### DISCUSSION TITLE
How AND_PERP strong is?

### DISCUSSION THREAD

**Discussion by miasik** [ljleb/sd-webui-neutral-prompt#58] [Q&A] [answered]
I have a promt describing a picture with a face with leaves in green palette. I'd like to use AND_PERP to change her eyes color to golden.
I've already tried several times to do it but failed.
Is there a way to debug my mistake?
Or can I make "AND_PERP golden eyes" stronger?

**Comment by ljleb** [comment, marked as answer]
`AND_PERP` will change the entire image. I think `<your prompt> AND_TOPK <your prompt again>, golden eyes` would work better. You will likely have a better result if you repeat your full prompt in different sub prompts. The reason is that each sub prompt tries to describe the entire picture, not just a specific feature.

An alternative solution could be to simply use prompt editing: `<your prompt> [:, golden eyes:0.4]`.

I just started working on #25, which is all that is missing to have the full SEGA implementation (semantic guidance). IIUC, SEGA would work well for your use case.

I intend to implement this in [prompt fusion](https://github.com/ljleb/prompt-fusion-extension) with the following syntax:
```
<your prompt> [ AND_TOPK <your prompt>, golden eyes : : 0.4 ]
```

So in other words, it will allow to dynamically change the composable diffusion prompts at runtime.

One optimization that needs to be done currently is simplifying the way prompts are written. Instead of having to type
```
<your prompt>
AND_TOPK <your prompt again>, golden eyes
```
it will be possible to type
```
PREPEND <your prompt>
AND_TOPK , golden eyes
```

or something like that.

**Reply by miasik** [reply]
Didn't think about TOPK because of lack of its examples. Will try it.
But what is PREPEND? I've never seen it before.
Thanks a lot!

**Reply by ljleb** [reply]
`PREPEND` is not a thing yet, but it will prepend a prompt to all other prompts at a given level. It's useful to prepend a prompt to other prompts since each prompt should describe the entire image every time.

**Reply by miasik** [reply]
Sorry, hard to get that. Could you please make an example for `PREPEND`?

**Reply by ljleb** [reply]
I will add examples to the readme once it is implemented, as it is not implemented yet.

example for hypothetical `PREPEND`:
```
PREPEND a nice chair
AND_PERP , mediocre ugly abstract art, low quality, lowres :-1
```

which would translate literally to:
```
a nice chair
AND_PERP a nice chair, mediocre ugly abstract art, low quality, lowres :-1
```

I'm suggesting this new keyword because I think it would make the extension easier to use correctly.

**Reply by miasik** [reply]
got it!
It's like a defining a variable for later using.
Super!
